src/scripts/install.sh

```diff
@@ -1,45 +1,82 @@
-downloadAndRun() {
-    url=$1
-    shift
+#!/usr/bin/env bash
+
+# Exit script if you try to use an uninitialized variable.
+set -o nounset
+
+# Exit script if a statement returns a non-true return value.
+set -o errexit
+
+# Use the error status of the first failure, rather than that of the last item in a pipeline.
+set -o pipefail
+
+sudoIfAvailable() {
     if [[ -x $(command -v sudo) ]]; then
-    curl -sfL $url | sudo -E bash -s -- "$@"
+    sudo -E bash "$@"
     else
-    curl -sfL $url | bash -s -- "$@"
+    bash "$@"
     fi
 }
 
+downloadAndRun() {
+    url=$1
+    shift
+    curl -sfL $url | sudoIfAvailable -s -- "$@"
+}
+
 os=$(uname)
+tmpdir=$(dirname "$(mktemp -u)")
+mpmbaseurl="https://www.mathworks.com/mpm"
 
-# installer does not support the Docker executor type on Linux
-if [[ $os = Linux ]] && awk -F/ '$2 == "docker"' /proc/self/cgroup | read -r; then
-    echo 'The Docker executor type is not supported.'
-    exit 1
+# install system dependencies
+if [[ $os = Linux ]]; then
+    # install MATLAB dependencies
+    release=$(echo "${PARAM_RELEASE}" | grep -ioE "(r[0-9]{4}[a-b]|latest)")
```

Comment:
Should this be handled in matlab-deps instead? Saves us from having to repeat it in all of our plugins and it may be possible that different updates require different deps.